### DISCUSSION TITLE
Docker compose with full stack

### DISCUSSION THREAD

**Discussion by ChronoDK** [Tobi-De/cookiecutter-fastapi#3] [Ideas]
Cool project here! Would be very nice if the project had a docker compose file with the full stack: mongo, redis etc. - that way I could just run that, and have a full setup going. Also, I would love if the frontend example was more than what it is: A log-in dialog to demonstrate fastapi-users auth.

Anyway, not sure if you are still working on this or if there is anything better out there. Just found this and tested it, and I hoped it had gotten me a bit further that it did.

**Comment by Tobi-De** [comment]
Thank you for your feedback, @ChronoDK!

I'm glad you find the project cool. Have you considered using Docker Compose for development or production or both ? Typically, I don't use Docker or Docker Compose for development because I find it too cumbersome and for production the dockerfile included in the repo is usually enough for me. However, I can certainly add some examples or details in the docs for people looking for a more complete solution. In the meantime, I've got some resources that might be useful if you're still figuring things out:

- https://github.com/tiangolo/blog-posts/tree/master/deploying-fastapi-apps-with-https-powered-by-traefik
- https://fastapi.tiangolo.com/deployment/
- https://testdriven.io/blog/fastapi-docker-traefik/

If you're looking to save yourself some headaches, I highly recommend using a PaaS (Platform as a Service)  solution if it's within your budget. Deployment can be tricky and there's always room for error. Here's a handy [list of PaaS](https://tobi-de.github.io/fuzzy-couscous/deployment/) options to consider.

Regarding templates, I completely agree that they're sorely needed! It's something I've been meaning to add but haven't gotten around to yet. Basic login, signup and home page templates are definitely on my to-do list.

I, for one, have not used fastapi recently, but I will gladly continue to maintain this cookiecutter.
In terms of better options, it ultimately depends on your specific needs and preferences. There are many good alternatives available, and I designed this project to provide a simple and straightforward way to recreate a Django-like experience using FastAPI. That being said, here are some additional options I can suggest:

- https://github.com/ycd/manage-fastapi
- https://github.com/s3rius/FastAPI-template
- https://github.com/rednafi/fastapi-nano
- https://github.com/fastapi-mvc/fastapi-mvc
- https://github.com/teamhide/fastapi-boilerplate
- https://github.com/gaganpreet/fastapi-starter

Thanks again for your interest in my project! Apologies  for the delayed response - I've got quite a few incomplete projects on the go at the moment :smile:
